bpfd/src/server/bpf.rs

```diff
@@ -100,64 +102,104 @@ impl<'a> BpfManager<'a> {
-            id,
-            ExtensionProgram {
-                path,
-                loader: Some(ext_loader),
-                current_position: None,
-                metadata: Metadata {
-                    priority,
-                    name: section_name,
-                    attached: false,
-                },
-                link: None,
-                owner,
-                proceed_on,
-            },
-        );
-        self.sort_extensions(&if_index);
-
-        let mut dispatcher_loader =
-            self.new_dispatcher(&if_index, next_available_id as u8, self.dispatcher_bytes)?;
-
-        // Keep old_links in scope until after this function exits to avoid dropping
-        // them before the new dispatcher is attached
-        let _old_links = self.attach_extensions(&if_index, &mut dispatcher_loader)?;
-        self.update_or_replace_dispatcher(iface.clone(), if_index, dispatcher_loader)?;
-        info!(
-            "Program added: {} programs attached to {}",
-            self.programs.get(&if_index).unwrap().len(),
-            &iface,
-        );
-        Ok(id)
+                self.programs.get_mut(&if_index).unwrap().insert(
+                    id,
+                    ExtensionProgram {
+                        path,
+                        loader: Some(ext_loader),
+                        current_position: None,
+                        metadata: Metadata {
+                            priority,
+                            name: section_name,
+                            attached: false,
+                        },
+                        link: None,
+                        owner,
+                        proceed_on,
+                    },
+                );
+                self.sort_extensions(&if_index);
+
+                let mut dispatcher_loader =
+                    self.new_dispatcher(&if_index, next_available_id as u8, self.dispatcher_bytes)?;
+
+                // Keep old_links in scope until after this function exits to avoid dropping
+                // them before the new dispatcher is attached
+                let _old_links = self.attach_extensions(&if_index, &mut dispatcher_loader)?;
+                self.update_or_replace_dispatcher(iface.clone(), if_index, dispatcher_loader)?;
+                info!(
+                    "Program added: {} programs attached to {}",
+                    self.programs.get(&if_index).unwrap().len(),
+                    &iface,
+                );
+                Ok(id)
+            }
+            1 => {
+                // error adding clsact to the interface if it is already added is harmless
+                // the full cleanup can be done with 'sudo tc qdisc del dev iface clsact'.
+                println!("Trace 03: {:#?}", &iface);
+                let _ = tc::qdisc_add_clsact(&iface);
+                println!("Trace 03.1: {:#?}", &iface);
+                let mut ext_loader = BpfLoader::new().load_file(path.clone())?;
```

Comment:
When I execute the following command:
`$ sudo ./target/debug/bpfctl load /home/afredette/bpf/accept-all.o -p tc_ingress -i vboxnet1 --priority 50 -s "ingress"`
I get the following error:
`Error: status: Aborted, message: "error parsing BPF object", details: [], metadata: MetadataMap { headers: {"content-type": "application/grpc", "date": "Mon, 22 Aug 2022 22:44:36 GMT", "content-length": "0"} }`

Reply:
The issue was that the section name needs to be named "classifier".